multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -97,55 +80,22 @@ func NewMemberClusterAnnounceReconciler(client client.Client, scheme *runtime.Sc
 	memberAnnounce := &multiclusterv1alpha1.MemberClusterAnnounce{}
-	err := r.Get(ctx, req.NamespacedName, memberAnnounce)
+	var err error
+	memberID := getIDFromName(req.Name)
+	err = r.Get(ctx, req.NamespacedName, memberAnnounce)
 	if err != nil {
-		if !errors.IsNotFound(err) {
-			// Cannot read the requested resource. Return error, so reconciliation will be retried.
-			return ctrl.Result{}, err
-		}
-		// If MemberClusterAnnounce is deleted, no need to process because RemoveMember must already
-		// be called.
+		// If MemberClusterAnnounce is deleted, no further processing is needed, as cleanup
+		// must have been done when the Finalizer was removed.
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	r.mapLock.Lock()
-	// If the member is not found in timerData, it means the member is not added to the ClusterSet or was deleted.
-	if data, ok := r.timerData[common.ClusterID(memberAnnounce.ClusterID)]; ok {
-		klog.V(2).InfoS("Reset lastUpdateTime", "cluster", memberAnnounce.ClusterID)
-		// Reset lastUpdateTime and connectedLeader for this member.
-		data.lastUpdateTime = time.Now()
-		if len(memberAnnounce.LeaderClusterID) == 0 {
-			data.leaderStatus.connectedLeader = v1.ConditionUnknown
-			data.leaderStatus.message = "Not connected to leader yet"
-			data.leaderStatus.reason = ""
-		} else {
-			data.leaderStatus.connectedLeader = v1.ConditionFalse
-			data.leaderStatus.message = fmt.Sprintf("Local cluster is not the leader of member: %v",
-				memberAnnounce.ClusterID)
-			data.leaderStatus.reason = ReasonNotLeader
-			// Check whether this local cluster is the leader for this member.
-			clusterClaim := &multiclusterv1alpha2.ClusterClaim{}
-			if err := r.Get(context.TODO(), types.NamespacedName{Name: multiclusterv1alpha2.WellKnownClusterClaimID, Namespace: req.Namespace}, clusterClaim); err == nil {
-				if clusterClaim.Value == memberAnnounce.LeaderClusterID {
-					data.leaderStatus.connectedLeader = v1.ConditionTrue
-					data.leaderStatus.message = fmt.Sprintf("Local cluster is the leader of member: %v",
-						memberAnnounce.ClusterID)
-					data.leaderStatus.reason = ReasonConnectedLeader
-				}
-			}
-			// If err != nil, probably ClusterClaims were deleted during the processing of MemberClusterAnnounce.
-			// Nothing to handle in this case and MemberClusterAnnounce will also be deleted soon.
-			// TODO: Add ClusterClaim webhook to make sure it cannot be deleted while ClusterSet is present.
-		}
-	}
-	r.mapLock.Unlock()
-
 	finalizer := fmt.Sprintf("%s/%s", MemberClusterAnnounceFinalizer, memberAnnounce.ClusterID)
 	if !memberAnnounce.DeletionTimestamp.IsZero() {
 		if err := r.removeMemberFromClusterSet(memberAnnounce); err != nil {
 			klog.ErrorS(err, "Failed to remove member cluster from the ClusterSet", "cluster", memberAnnounce.ClusterID)
 			return ctrl.Result{}, err
 		}
+		r.removeMemberStatus(memberID)
```

Comment:
Question - should we call removeMemberStatus() after removing Finalizer? E.g. is it possible that a member cluster can clean the DeletionTimestamp and continue to update/reuse the Announce?

Reply:
From member's perspective, the MemberClusterAnnounce will be deleted when ClusterSet is deleted, so it shouldn't be able to talk to leader any more.

Reply:
>  E.g. is it possible that a member cluster can clean the DeletionTimestamp and continue to update/reuse the Announce?

Even a member cluster should never do that, could a user manually do that?

Reply:
I don't think DeletionTimestamp is mutable, here is the comment of `DeletionTimestamp`.
```
// Once the deletionTimestamp is set, this value may not be unset or be set further into the
// future, although it may be shortened or the resource may be deleted prior to this time.
```

Reply:
Ok.